### PR TITLE
Fix insert link bug and add notifications on failures

### LINKED ISSUE #17: Create node commands makes empty links and it doesn't follow me to real Child or Brother/Sister node

### PR DESCRIPTION
**Pull request by pauloday**
This fixes a bug where the add Zettel link command wouldn't work on my notes. I'm not sure why it was working on my test archive but not my actual one - it shouldn't have worked on either. Anyway it works on both now. I also added some notifications when things fail, so you don't get nothing happening silently when it can't get an id from the filename or something.

Fixes #17.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -308,15 +310,15 @@ export default class NewZettel extends Plugin {
...
     }
   }
 
   async renameZettel(id: string, toId: string) {
     const sep = this.settings.separator;
-    let zettel = this.app.vault
+    const zettel = this.app.vault
       .getMarkdownFiles()
-      .filter((file) => this.fileToId(file.basename) == id)
+      .filter(file => this.fileToId(file.basename) === id)
```
Comment (brannonh) [inline src/main.ts:321]: Do you have an example case where `==` causes problems?

**Review by brannonh — COMMENTED** [review]
I know it's been awhile, but just got access to start moving things along here. Are you still interested in merging this PR? If so, please address merge conflicts and my question in `main.ts`.

If you're not interested in being involved, please let us know.

Thanks!

**Comment by Eforen** [comment]
@brannonh how would I go about pulling this down and fixing the conflict? Do I fork the repo or his repo or both or what?

I have not delt with pull requests in 2 repos I don't have permissions on before...

I see this as a kinda critical issue.

**Comment by brannonh** [comment]
> @brannonh how would I go about pulling this down and fixing the conflict? Do I fork the repo or his repo or both or what?
> 
> I have not delt with pull requests in 2 repos I don't have permissions on before...
> 
> I see this as a kinda critical issue.

Honestly, I have never done that either. Let me see if I can resolve the conflicts real quick.

File: src/main.ts [hunk 2]
```diff
@@ -481,11 +493,10 @@ export default class NewZettel extends Plugin {
...
 
   getZettels(): TFile[] {
+    const fileToId = (file: TFile) => this.fileToId(file.basename)
     return this.app.vault.getMarkdownFiles().filter(file => {
-      const ignore = /^[_layouts|templates|scripts]/;
-      return
-        this.fileToId(file.basename) !== '' &&
-        file.path.match(ignore) == null;
+      const ignore = !file.path.match(/^[_layouts|templates|scripts]/);
+      return ignore && fileToId(file) !== '';
```
Comment (brannonh) [inline src/main.ts:504]: I think there is a missing `this.` here.

**Comment by brannonh** [comment]
@Eforen I was able to resolve the conflict. Looks like there is a missing `this.` in one line, but I can fix that right after the merge. I'll try to get this merged in the next couple of days.

**Comment by brannonh** [comment]
Fixes #17.

File: src/main.ts (hunk 2 above)
Comment (Eforen) [reply src/main.ts:504]: I agree

**Pull request merged**